test/remote/gateways/remote_priority_test.rb

```diff
@@ -0,0 +1,352 @@
+    assert_equal 'Declined', response.params['status']
+  end
+
+  # Authorize tests
+  def test_successful_authorize
+    response = @gateway.authorize(@amount_purchase, @credit_card_purchase_success, @option_spr)
+    assert_success response
+    assert_equal 'Approved', response.params['status']
+  end
+
+  # Invalid card number
+  def test_failed_authorize
+    response = @gateway.authorize(@amount_purchase, @credit_card_purchase_fail_invalid_number, @option_spr)
+    assert_failure response
+
+    assert_equal 'Invalid card number', response.params['authMessage']
+    assert_equal 'Declined', response.params['status']
+  end
+
+  # Missing card number month
+  def test_failed_authorize_missing_card_month
+    response = @gateway.authorize(@amount_purchase, @credit_card_purchase_fail_missing_month, @option_spr)
+    assert_failure response
+
+    assert_equal 'ValidationError', response.params['errorCode']
+    assert_equal 'Validation error happened', response.params['message']
+    assert_equal 'Missing expiration month and / or year', response.params['details'][0]
+  end
+
+  # Missing card verification number
+  def test_failed_authorize_missing_card_verification_number
+    response = @gateway.authorize(@amount_purchase, @credit_card_purchase_fail_missing_verification, @option_spr)
+    assert_failure response
+
+    assert_equal 'CVV is required based on merchant fraud settings', response.params['authMessage']
+    assert_equal 'Declined', response.params['status']
+  end
+
+  # Capture tests
+  def test_successful_capture
+    auth_obj = @gateway.authorize(@amount_authorize, @credit_card_purchase_success, @option_spr)
+    assert_success auth_obj
+    # add auth code to options
+    @option_spr.update(authCode: auth_obj.params['authCode'])
+
+    capture = @gateway.capture(@amount_authorize, auth_obj.authorization, @option_spr)
+    assert_success capture
+    assert_equal 'Approved', capture.params['authMessage']
+    assert_equal 'Approved', capture.params['status']
+  end
+
+  # Invalid authorization and null auth code
+  def test_failed_capture
+    # add auth code to options
+    @option_spr.update(authCode: '12345')
+    capture = @gateway.capture(@amount_authorize, 'bogus', @option_spr)
+    assert_failure capture
+
+    assert_equal 'Original Transaction Not Found', capture.params['authMessage']
+    assert_equal 'Declined', capture.params['status']
+  end
+
+  # Void tests
+  # Batch status is by default is set to Open wehn Sale transaction is created
+  def test_successful_void_batch_open
+    response = @gateway.purchase(@amount_purchase, @credit_card_purchase_success, @option_spr)
+    assert_success response
+
+    # check is this transaction associated batch is "Closed".
+    batch_check = @gateway.get_payment_status(response.params['batchId'], @option_spr)
+    # if batch Open then fail test. Batch must be closed to perform a Refund
```

Comment:
I have two questions here:
1. Are we just at the mercy of the sandbox environment regarding whether or not the batch is "Open" or "Closed"?
2. If the batch status is "Closed" does the `void` call still succeed?

Reply:
@dsmcclain yeah as far as I can tell the only way to get the payment status for a batch is to call this method and pull it from them.

2. It looks like from https://developer.mxmerchant.com/docs/making-a-full-refund-or-void-a-transaction that if a batch is closed and you void an associated transaction in it, it'll open up a refund for that transaction. I did test it and looks like the status is now `pending` for the batch so I'm going to assume that is the intended flow and put in a test to reflect that.